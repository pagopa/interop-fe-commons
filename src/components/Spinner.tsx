import React from 'react'
import { CircularProgress, Stack, type StackProps, type SxProps, Typography } from '@mui/material'

type SpinnerProps = {
  label?: string | null
  direction?: StackProps['direction']
  sx?: SxProps
}

export function Spinner({ label, direction = 'column', sx }: SpinnerProps) {
  return (
    <Stack sx={sx} direction={direction} alignItems="center" justifyContent="center" spacing={2}>
      <CircularProgress />
      {label && (
        <Typography component="p" variant="body2" fontWeight={700} color="primary" sx={{ mb: 0 }}>
          {label}
        </Typography>
      )}
    </Stack>
  )
}
