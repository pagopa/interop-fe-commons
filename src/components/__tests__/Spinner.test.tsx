import React from 'react'
import { Spinner } from '../Spinner'
import renderer from 'react-test-renderer'

describe('Spinner component', () => {
  it('Renders correctly with label and column direction', () => {
    const tree = renderer.create(<Spinner label={'test-title'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders correctly without label and column direction', () => {
    const tree = renderer.create(<Spinner />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders correctly with label and row direction', () => {
    const tree = renderer.create(<Spinner direction="row" label={'test-title'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders correctly without label and row direction', () => {
    const tree = renderer.create(<Spinner direction="row" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
