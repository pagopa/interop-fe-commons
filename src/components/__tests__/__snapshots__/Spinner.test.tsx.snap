// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner component > Renders correctly with label and column direction 1`] = `
<div
  className="MuiStack-root css-10y7f4i-MuiStack-root"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={{}}
      />
    </svg>
  </span>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1mzt6m4-MuiTypography-root"
  >
    test-title
  </p>
</div>
`;

exports[`Spinner component > Renders correctly with label and row direction 1`] = `
<div
  className="MuiStack-root css-1m9ibdo-MuiStack-root"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={{}}
      />
    </svg>
  </span>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1mzt6m4-MuiTypography-root"
  >
    test-title
  </p>
</div>
`;

exports[`Spinner component > Renders correctly without label and column direction 1`] = `
<div
  className="MuiStack-root css-10y7f4i-MuiStack-root"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={{}}
      />
    </svg>
  </span>
</div>
`;

exports[`Spinner component > Renders correctly without label and row direction 1`] = `
<div
  className="MuiStack-root css-1m9ibdo-MuiStack-root"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={{}}
      />
    </svg>
  </span>
</div>
`;
